lib/src/rules/avoid_dynamic_calls.dart

```diff
@@ -193,6 +195,11 @@ class _Visitor extends SimpleAstVisitor<void> {
 
   @override
   void visitMethodInvocation(MethodInvocation node) {
+    final methodName = node.methodName.name;
+    if (methodName == 'noSuchMethod' || methodName == 'toString') {
```

Comment:
Also check the number of arguments. In order to be ignored by the lint, there should be 1 positional parameter for `noSuchMethod` and 0 for `toString`, and in both cases no named parameters.